maven-scm-providers/maven-scm-providers-git/maven-scm-provider-jgit/src/main/java/org/apache/maven/scm/provider/git/jgit/command/JGitUtils.java

```diff
@@ -527,4 +529,61 @@ public static List<String> getTags( Repository repo, RevCommit commit ) throws I
         }
         return result;
     }
+
+    /**
+     * Remove all files in the given fileSet to the repository.
+     *
+     * @param git     the repo to remove the files from
+     * @param fileSet the set of files within the workspace, the files are removed
+     *                relative to the basedir of this fileset
+     * @return a list of removed files
+     * @throws GitAPIException
+     * @throws NoFilepatternException
+     */
+    public static List<ScmFile> removeAllFiles( Git git, ScmFileSet fileSet )
+        throws GitAPIException, NoFilepatternException
+    {
+        URI baseUri = fileSet.getBasedir().toURI();
+        RmCommand remove = git.rm();
+        for ( File file : fileSet.getFileList() )
+        {
+            if ( !file.isAbsolute() )
+            {
+                file = new File( fileSet.getBasedir().getPath(), file.getPath() );
+            }
+
+            if ( file.exists() )
+            {
+                URI workingCopyRootUri = git.getRepository().getWorkTree().toURI();
+                String path = FilenameUtils.normalizeFilename( relativize( workingCopyRootUri, file ) );
+                remove.addFilepattern( path );
+            }
+        }
+        remove.call();
+
+        Status status = git.status().call();
+
+        Set<String> allInIndex = new HashSet<String>();
+        allInIndex.addAll( status.getRemoved() );
+        List<ScmFile> removedFiles = new ArrayList<ScmFile>( allInIndex.size() );
+
+        // rewrite all detected files to now have status 'checked_in'
+        for ( String entry : allInIndex )
+        {
+            ScmFile scmFile = new ScmFile( entry, ScmFileStatus.DELETED );
+
+            // if a specific fileSet is given, we have to check if the file is
+            // really tracked
+            for ( Iterator<File> itfl = fileSet.getFileList().iterator(); itfl.hasNext(); )
+            {
+                String path = FilenameUtils.normalizeFilename( relativize( baseUri, itfl.next() ) );
+                if ( path.equals( FilenameUtils.normalizeFilename( scmFile.getPath() ) ) )
```

Comment:
this logic is IMHO incorrect, as `scmFile` has repository relative path names (already with forward slashes) while `path` is relative to basedir.